// @mondal/org-chart/esm5/lib/employee.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
/**
 * @record
 */
export function IEmployee() { }
if (false) {
    /** @type {?} */
    IEmployee.prototype.name;
    /** @type {?} */
    IEmployee.prototype.cssClass;
    /** @type {?} */
    IEmployee.prototype.imageUrl;
    /** @type {?} */
    IEmployee.prototype.designation;
    /** @type {?} */
    IEmployee.prototype.subordinates;
}
var Employee = /** @class */ (function () {
    function Employee(orgStructure, manager) {
        var _this = this;
        this.manager = manager;
        var _a = tslib_1.__read(orgStructure), name = _a[0], reports = _a.slice(1);
        this.name = name.split('(')[0].trim();
        /** @type {?} */
        var desigMatch = name.match(/\(([^)]+)\)/);
        this.designation = desigMatch && desigMatch[1].trim();
        this.subordinates = reports.map((/**
         * @param {?} r
         * @return {?}
         */
        function (r) { return r.substring(1); }))
            .reduce((/**
         * @param {?} previous
         * @param {?} current
         * @return {?}
         */
        function (previous, current) {
            if (!current.startsWith(' ')) {
                previous.push([]);
            }
            previous[previous.length - 1].push(current);
            return previous;
        }), (/** @type {?} */ ([])))
            .map((/**
         * @param {?} r
         * @return {?}
         */
        function (r) { return new Employee(r, _this); }));
    }
    return Employee;
}());
export { Employee };
if (false) {
    /** @type {?} */
    Employee.prototype.name;
    /** @type {?} */
    Employee.prototype.cssClass;
    /** @type {?} */
    Employee.prototype.imageUrl;
    /** @type {?} */
    Employee.prototype.designation;
    /** @type {?} */
    Employee.prototype.subordinates;
    /** @type {?} */
    Employee.prototype.manager;
}
