// @mondal/org-chart/esm5/lib/employee/employee.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, EventEmitter, Input, Output } from '@angular/core';
var EmployeeComponent = /** @class */ (function () {
    function EmployeeComponent() {
        this.hasManager = false;
        this.direction = 'vertical';
        this.itemClick = new EventEmitter();
    }
    EmployeeComponent.decorators = [
        { type: Component, args: [{
                    selector: 'mui-employee',
                    template: "<div ngClass=\"mui-oc-connector-{{direction}}\" class=\"mui-oc-border\" *ngIf=\"hasManager\"></div>\n<div [ngClass]=\"employee?.cssClass\" class=\"mui-oc-box mui-oc-border mui-oc-background\"\n  (click)=\"itemClick.emit(employee)\">\n  <div *ngIf=\"employee?.imageUrl\" class=\"mui-oc-image mui-oc-border\"\n    [ngStyle]=\"{ 'background-image': 'url(\\''+ employee?.imageUrl+'\\')'}\"></div>\n  <div>\n    <div class=\"mui-oc-name\">{{employee?.name}}</div>\n    <div class=\"mui-oc-designation\">{{employee?.designation}}</div>\n  </div>\n</div>\n<div ngClass=\"mui-oc-connector-{{direction}}\" class=\"mui-oc-border\" *ngIf=\"employee?.subordinates?.length\"></div>",
                    styles: [":host{display:flex;align-items:center}.mui-oc-box{cursor:pointer;display:flex;align-items:center}"]
                }] }
    ];
    EmployeeComponent.propDecorators = {
        employee: [{ type: Input }],
        hasManager: [{ type: Input }],
        direction: [{ type: Input }],
        itemClick: [{ type: Output }]
    };
    return EmployeeComponent;
}());
export { EmployeeComponent };
if (false) {
    /** @type {?} */
    EmployeeComponent.prototype.employee;
    /** @type {?} */
    EmployeeComponent.prototype.hasManager;
    /** @type {?} */
    EmployeeComponent.prototype.direction;
    /** @type {?} */
    EmployeeComponent.prototype.itemClick;
}
