// @mondal/org-chart/esm5/lib/org-chart/org-chart.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, EventEmitter, HostBinding, Input, Output } from '@angular/core';
var OrgChartComponent = /** @class */ (function () {
    function OrgChartComponent() {
        this.hasManager = false;
        this.direction = 'vertical';
        this.itemClick = new EventEmitter();
    }
    Object.defineProperty(OrgChartComponent.prototype, "hostClass", {
        get: /**
         * @return {?}
         */
        function () {
            return this.direction === 'vertical' ? 'column' : '';
        },
        enumerable: true,
        configurable: true
    });
    OrgChartComponent.decorators = [
        { type: Component, args: [{
                    selector: 'mui-org-chart',
                    template: "<mui-employee *ngIf=\"topEmployee\" ngClass=\"mui-oc-{{direction}}\" [employee]=\"topEmployee\" [hasManager]=\"hasManager\"\n  [direction]=\"direction\" (itemClick)=\"itemClick.emit($event)\">\n</mui-employee>\n<div *ngIf=\"topEmployee?.subordinates?.length\" ngClass=\"mui-oc-reports-{{direction}}\" class=\"mui-oc-reports\">\n  <ng-container *ngFor=\"let employee of topEmployee?.subordinates; first as isFirst; last as isLast\">\n    <div ngClass=\"mui-oc-org-container-{{direction}}\" class=\"mui-oc-org-container\">\n      <div ngClass=\"mui-oc-connector-container-{{direction}}\" class=\"mui-oc-connector-container\">\n        <div class=\"mui-oc-connector mui-oc-border\" [style.border-color]=\"isFirst?'transparent':''\"></div>\n        <div class=\"mui-oc-border\"></div>\n        <div class=\"mui-oc-connector mui-oc-border\" [style.border-color]=\"isLast?'transparent':''\"></div>\n      </div>\n      <mui-org-chart [topEmployee]=\"employee\" [hasManager]=\"true\" [direction]=\"direction\"\n        (itemClick)=\"itemClick.emit($event)\">\n      </mui-org-chart>\n    </div>\n  </ng-container>\n</div>",
                    styles: [":host{display:flex;align-items:center;flex:1}.mui-oc-vertical{flex-direction:column}.mui-oc-org-container{display:flex}.mui-oc-org-container-vertical{flex-direction:column}.mui-oc-connector{flex:1}.mui-oc-connector-container{display:flex}.mui-oc-connector-container-horizontal{flex-direction:column}.mui-oc-reports{display:flex;flex:1}.mui-oc-reports-horizontal{flex-direction:column}"]
                }] }
    ];
    OrgChartComponent.propDecorators = {
        topEmployee: [{ type: Input }],
        hasManager: [{ type: Input }],
        direction: [{ type: Input }],
        itemClick: [{ type: Output }],
        hostClass: [{ type: HostBinding, args: ['style.flex-direction',] }]
    };
    return OrgChartComponent;
}());
export { OrgChartComponent };
if (false) {
    /** @type {?} */
    OrgChartComponent.prototype.topEmployee;
    /** @type {?} */
    OrgChartComponent.prototype.hasManager;
    /** @type {?} */
    OrgChartComponent.prototype.direction;
    /** @type {?} */
    OrgChartComponent.prototype.itemClick;
}
