// @mondal/org-chart/esm2015/lib/employee.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/**
 * @record
 */
export function IEmployee() { }
if (false) {
    /** @type {?} */
    IEmployee.prototype.name;
    /** @type {?} */
    IEmployee.prototype.cssClass;
    /** @type {?} */
    IEmployee.prototype.imageUrl;
    /** @type {?} */
    IEmployee.prototype.designation;
    /** @type {?} */
    IEmployee.prototype.subordinates;
}
export class Employee {
    /**
     * @param {?} orgStructure
     * @param {?=} manager
     */
    constructor(orgStructure, manager) {
        this.manager = manager;
        const [name, ...reports] = orgStructure;
        this.name = name.split('(')[0].trim();
        /** @type {?} */
        const desigMatch = name.match(/\(([^)]+)\)/);
        this.designation = desigMatch && desigMatch[1].trim();
        this.subordinates = reports.map((/**
         * @param {?} r
         * @return {?}
         */
        r => r.substring(1)))
            .reduce((/**
         * @param {?} previous
         * @param {?} current
         * @return {?}
         */
        (previous, current) => {
            if (!current.startsWith(' ')) {
                previous.push([]);
            }
            previous[previous.length - 1].push(current);
            return previous;
        }), (/** @type {?} */ ([])))
            .map((/**
         * @param {?} r
         * @return {?}
         */
        r => new Employee(r, this)));
    }
}
if (false) {
    /** @type {?} */
    Employee.prototype.name;
    /** @type {?} */
    Employee.prototype.cssClass;
    /** @type {?} */
    Employee.prototype.imageUrl;
    /** @type {?} */
    Employee.prototype.designation;
    /** @type {?} */
    Employee.prototype.subordinates;
    /** @type {?} */
    Employee.prototype.manager;
}
