// @mondal/org-chart/__ivy_ngcc__/fesm2015/mondal-org-chart.js
import { CommonModule } from '@angular/common';
import { EventEmitter, Component, Input, Output, HostBinding, NgModule } from '@angular/core';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';

function EmployeeComponent_div_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "div", 5);
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵpropertyInterpolate1("ngClass", "mui-oc-connector-", ctx_r0.direction, "");
} }
const _c0 = function (a0) { return { "background-image": a0 }; };
function EmployeeComponent_div_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "div", 6);
} if (rf & 2) {
    const ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("ngStyle", ɵngcc0.ɵɵpureFunction1(1, _c0, "url('" + (ctx_r1.employee == null ? null : ctx_r1.employee.imageUrl) + "')"));
} }
function EmployeeComponent_div_8_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "div", 5);
} if (rf & 2) {
    const ctx_r2 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵpropertyInterpolate1("ngClass", "mui-oc-connector-", ctx_r2.direction, "");
} }
function OrgChartComponent_mui_employee_0_Template(rf, ctx) { if (rf & 1) {
    const _r3 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "mui-employee", 2);
    ɵngcc0.ɵɵlistener("itemClick", function OrgChartComponent_mui_employee_0_Template_mui_employee_itemClick_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r3); const ctx_r2 = ɵngcc0.ɵɵnextContext(); return ctx_r2.itemClick.emit($event); });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵpropertyInterpolate1("ngClass", "mui-oc-", ctx_r0.direction, "");
    ɵngcc0.ɵɵproperty("employee", ctx_r0.topEmployee)("hasManager", ctx_r0.hasManager)("direction", ctx_r0.direction);
} }
function OrgChartComponent_div_1_ng_container_1_Template(rf, ctx) { if (rf & 1) {
    const _r9 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵelementStart(1, "div", 5);
    ɵngcc0.ɵɵelementStart(2, "div", 6);
    ɵngcc0.ɵɵelement(3, "div", 7);
    ɵngcc0.ɵɵelement(4, "div", 8);
    ɵngcc0.ɵɵelement(5, "div", 7);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(6, "mui-org-chart", 9);
    ɵngcc0.ɵɵlistener("itemClick", function OrgChartComponent_div_1_ng_container_1_Template_mui_org_chart_itemClick_6_listener($event) { ɵngcc0.ɵɵrestoreView(_r9); const ctx_r8 = ɵngcc0.ɵɵnextContext(2); return ctx_r8.itemClick.emit($event); });
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    const employee_r5 = ctx.$implicit;
    const isFirst_r6 = ctx.first;
    const isLast_r7 = ctx.last;
    const ctx_r4 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵpropertyInterpolate1("ngClass", "mui-oc-org-container-", ctx_r4.direction, "");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵpropertyInterpolate1("ngClass", "mui-oc-connector-container-", ctx_r4.direction, "");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵstyleProp("border-color", isFirst_r6 ? "transparent" : "");
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵstyleProp("border-color", isLast_r7 ? "transparent" : "");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("topEmployee", employee_r5)("hasManager", true)("direction", ctx_r4.direction);
} }
function OrgChartComponent_div_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 3);
    ɵngcc0.ɵɵtemplate(1, OrgChartComponent_div_1_ng_container_1_Template, 7, 9, "ng-container", 4);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵpropertyInterpolate1("ngClass", "mui-oc-reports-", ctx_r1.direction, "");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r1.topEmployee == null ? null : ctx_r1.topEmployee.subordinates);
} }
class EmployeeComponent {
    constructor() {
        this.hasManager = false;
        this.direction = 'vertical';
        this.itemClick = new EventEmitter();
    }
}
EmployeeComponent.ɵfac = function EmployeeComponent_Factory(t) { return new (t || EmployeeComponent)(); };
EmployeeComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: EmployeeComponent, selectors: [["mui-employee"]], inputs: { hasManager: "hasManager", direction: "direction", employee: "employee" }, outputs: { itemClick: "itemClick" }, decls: 9, vars: 6, consts: [["class", "mui-oc-border", 3, "ngClass", 4, "ngIf"], [1, "mui-oc-box", "mui-oc-border", "mui-oc-background", 3, "ngClass", "click"], ["class", "mui-oc-image mui-oc-border", 3, "ngStyle", 4, "ngIf"], [1, "mui-oc-name"], [1, "mui-oc-designation"], [1, "mui-oc-border", 3, "ngClass"], [1, "mui-oc-image", "mui-oc-border", 3, "ngStyle"]], template: function EmployeeComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtemplate(0, EmployeeComponent_div_0_Template, 1, 1, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "div", 1);
        ɵngcc0.ɵɵlistener("click", function EmployeeComponent_Template_div_click_1_listener() { return ctx.itemClick.emit(ctx.employee); });
        ɵngcc0.ɵɵtemplate(2, EmployeeComponent_div_2_Template, 1, 3, "div", 2);
        ɵngcc0.ɵɵelementStart(3, "div");
        ɵngcc0.ɵɵelementStart(4, "div", 3);
        ɵngcc0.ɵɵtext(5);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(6, "div", 4);
        ɵngcc0.ɵɵtext(7);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(8, EmployeeComponent_div_8_Template, 1, 1, "div", 0);
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngIf", ctx.hasManager);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngClass", ctx.employee == null ? null : ctx.employee.cssClass);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.employee == null ? null : ctx.employee.imageUrl);
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵtextInterpolate(ctx.employee == null ? null : ctx.employee.name);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵtextInterpolate(ctx.employee == null ? null : ctx.employee.designation);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.employee == null ? null : ctx.employee.subordinates == null ? null : ctx.employee.subordinates.length);
    } }, directives: [ɵngcc1.NgIf, ɵngcc1.NgClass, ɵngcc1.NgStyle], styles: ["[_nghost-%COMP%]{display:flex;align-items:center}.mui-oc-box[_ngcontent-%COMP%]{cursor:pointer;display:flex;align-items:center}"] });
EmployeeComponent.propDecorators = {
    employee: [{ type: Input }],
    hasManager: [{ type: Input }],
    direction: [{ type: Input }],
    itemClick: [{ type: Output }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(EmployeeComponent, [{
        type: Component,
        args: [{
                selector: 'mui-employee',
                template: "<div ngClass=\"mui-oc-connector-{{direction}}\" class=\"mui-oc-border\" *ngIf=\"hasManager\"></div>\n<div [ngClass]=\"employee?.cssClass\" class=\"mui-oc-box mui-oc-border mui-oc-background\"\n  (click)=\"itemClick.emit(employee)\">\n  <div *ngIf=\"employee?.imageUrl\" class=\"mui-oc-image mui-oc-border\"\n    [ngStyle]=\"{ 'background-image': 'url(\\''+ employee?.imageUrl+'\\')'}\"></div>\n  <div>\n    <div class=\"mui-oc-name\">{{employee?.name}}</div>\n    <div class=\"mui-oc-designation\">{{employee?.designation}}</div>\n  </div>\n</div>\n<div ngClass=\"mui-oc-connector-{{direction}}\" class=\"mui-oc-border\" *ngIf=\"employee?.subordinates?.length\"></div>",
                styles: [":host{display:flex;align-items:center}.mui-oc-box{cursor:pointer;display:flex;align-items:center}"]
            }]
    }], function () { return []; }, { hasManager: [{
            type: Input
        }], direction: [{
            type: Input
        }], itemClick: [{
            type: Output
        }], employee: [{
            type: Input
        }] }); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class OrgChartComponent {
    constructor() {
        this.hasManager = false;
        this.direction = 'vertical';
        this.itemClick = new EventEmitter();
    }
    /**
     * @return {?}
     */
    get hostClass() {
        return this.direction === 'vertical' ? 'column' : '';
    }
}
OrgChartComponent.ɵfac = function OrgChartComponent_Factory(t) { return new (t || OrgChartComponent)(); };
OrgChartComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: OrgChartComponent, selectors: [["mui-org-chart"]], hostVars: 2, hostBindings: function OrgChartComponent_HostBindings(rf, ctx) { if (rf & 2) {
        ɵngcc0.ɵɵstyleProp("flex-direction", ctx.hostClass);
    } }, inputs: { hasManager: "hasManager", direction: "direction", topEmployee: "topEmployee" }, outputs: { itemClick: "itemClick" }, decls: 2, vars: 2, consts: [[3, "ngClass", "employee", "hasManager", "direction", "itemClick", 4, "ngIf"], ["class", "mui-oc-reports", 3, "ngClass", 4, "ngIf"], [3, "ngClass", "employee", "hasManager", "direction", "itemClick"], [1, "mui-oc-reports", 3, "ngClass"], [4, "ngFor", "ngForOf"], [1, "mui-oc-org-container", 3, "ngClass"], [1, "mui-oc-connector-container", 3, "ngClass"], [1, "mui-oc-connector", "mui-oc-border"], [1, "mui-oc-border"], [3, "topEmployee", "hasManager", "direction", "itemClick"]], template: function OrgChartComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtemplate(0, OrgChartComponent_mui_employee_0_Template, 1, 4, "mui-employee", 0);
        ɵngcc0.ɵɵtemplate(1, OrgChartComponent_div_1_Template, 2, 2, "div", 1);
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngIf", ctx.topEmployee);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.topEmployee == null ? null : ctx.topEmployee.subordinates == null ? null : ctx.topEmployee.subordinates.length);
    } }, directives: [ɵngcc1.NgIf, EmployeeComponent, ɵngcc1.NgClass, ɵngcc1.NgForOf, OrgChartComponent], styles: ["[_nghost-%COMP%]{display:flex;align-items:center;flex:1}.mui-oc-vertical[_ngcontent-%COMP%]{flex-direction:column}.mui-oc-org-container[_ngcontent-%COMP%]{display:flex}.mui-oc-org-container-vertical[_ngcontent-%COMP%]{flex-direction:column}.mui-oc-connector[_ngcontent-%COMP%]{flex:1}.mui-oc-connector-container[_ngcontent-%COMP%]{display:flex}.mui-oc-connector-container-horizontal[_ngcontent-%COMP%]{flex-direction:column}.mui-oc-reports[_ngcontent-%COMP%]{display:flex;flex:1}.mui-oc-reports-horizontal[_ngcontent-%COMP%]{flex-direction:column}"] });
OrgChartComponent.propDecorators = {
    topEmployee: [{ type: Input }],
    hasManager: [{ type: Input }],
    direction: [{ type: Input }],
    itemClick: [{ type: Output }],
    hostClass: [{ type: HostBinding, args: ['style.flex-direction',] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(OrgChartComponent, [{
        type: Component,
        args: [{
                selector: 'mui-org-chart',
                template: "<mui-employee *ngIf=\"topEmployee\" ngClass=\"mui-oc-{{direction}}\" [employee]=\"topEmployee\" [hasManager]=\"hasManager\"\n  [direction]=\"direction\" (itemClick)=\"itemClick.emit($event)\">\n</mui-employee>\n<div *ngIf=\"topEmployee?.subordinates?.length\" ngClass=\"mui-oc-reports-{{direction}}\" class=\"mui-oc-reports\">\n  <ng-container *ngFor=\"let employee of topEmployee?.subordinates; first as isFirst; last as isLast\">\n    <div ngClass=\"mui-oc-org-container-{{direction}}\" class=\"mui-oc-org-container\">\n      <div ngClass=\"mui-oc-connector-container-{{direction}}\" class=\"mui-oc-connector-container\">\n        <div class=\"mui-oc-connector mui-oc-border\" [style.border-color]=\"isFirst?'transparent':''\"></div>\n        <div class=\"mui-oc-border\"></div>\n        <div class=\"mui-oc-connector mui-oc-border\" [style.border-color]=\"isLast?'transparent':''\"></div>\n      </div>\n      <mui-org-chart [topEmployee]=\"employee\" [hasManager]=\"true\" [direction]=\"direction\"\n        (itemClick)=\"itemClick.emit($event)\">\n      </mui-org-chart>\n    </div>\n  </ng-container>\n</div>",
                styles: [":host{display:flex;align-items:center;flex:1}.mui-oc-vertical{flex-direction:column}.mui-oc-org-container{display:flex}.mui-oc-org-container-vertical{flex-direction:column}.mui-oc-connector{flex:1}.mui-oc-connector-container{display:flex}.mui-oc-connector-container-horizontal{flex-direction:column}.mui-oc-reports{display:flex;flex:1}.mui-oc-reports-horizontal{flex-direction:column}"]
            }]
    }], function () { return []; }, { hasManager: [{
            type: Input
        }], direction: [{
            type: Input
        }], itemClick: [{
            type: Output
        }], hostClass: [{
            type: HostBinding,
            args: ['style.flex-direction']
        }], topEmployee: [{
            type: Input
        }] }); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class OrgChartModule {
}
OrgChartModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: OrgChartModule });
OrgChartModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function OrgChartModule_Factory(t) { return new (t || OrgChartModule)(); }, imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(OrgChartModule, { declarations: function () { return [OrgChartComponent,
        EmployeeComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [OrgChartComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(OrgChartModule, [{
        type: NgModule,
        args: [{
                declarations: [OrgChartComponent, EmployeeComponent],
                imports: [
                    CommonModule
                ],
                exports: [OrgChartComponent]
            }]
    }], null, null); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

export { OrgChartModule, OrgChartComponent as ɵa, EmployeeComponent as ɵb };
